crates/sui-indexer-alt-jsonrpc/src/metrics/middleware.rs

```diff
@@ -42,6 +42,8 @@ pin_project! {
     pub(crate) struct MetricsFuture<'a, F> {
         metrics: Option<RequestMetrics>,
         method: Cow<'a, str>,
+        // RPC request params for logging
+        params: String,
```

Comment:
You should be able to store this as a reference to the underlying string:

```suggestion
        params: &'a str,
```